### PR TITLE
Include all audio before the response reflex

### PR DESCRIPTION
**Pull request by choombaa**
Right now, words spoken before the response reflex are being cut off due to buffering. I added a mutex that only allows the responseReflexEventHandler to fire after the next transcription.

One tradeoff here is that excess audio may be captured, but the user will usually be done speaking.

### CODE REVIEW

**Comment by ghost** [comment]
Won't this lead to responses being delayed a lot?

**Comment by choombaa** [comment]
That can be true, but I think that in the average case it will be beneficial. Subjectively, the responses have been much better in my testing.

Here are potential scenarios when the user says a sentence with an important word at the end, e.g. "What is your favorite brand of coffee?"

Status Quo (good case): "coffee" is included in the first transcription, no other words are required, the empty space is added to the cut transcription.

Status Quo (bad case): "Coffee" is cut off (response reflex triggers before transcription is complete). The LLM does not know the subject of the sentence, and talks about elephants instead of coffee. "Coffee" is added to the cut transcription, confusing the LLM later on.

After PR (good case): "Coffee" is cut off, but is included since we wait for the next transcription to finish. The LLM's response is good and on-topic, but this has cost us ~1 second of latency.

After PR (bad case): "Coffee" is not cut off, there is no need to wait for the transcription to finish, but we wait anyway. The LLM's response is good and on-topic, but this has cost us ~1 second of latency.

After PR (another good case): A large chunk of the sentence has been cut off, and "brand of coffee" is included in the next transcription. We still wait for ~1 second, but we have "wasted" much less of it, because we needed those words.

**Comment by yacineMTB** [comment]
thanks choombs
this is good, especially considering the fact that i am assuming whisper latency will go to 0 once all the ggml merges / optimizations make it there

File: index.ts [hunk 1]
```diff
@@ -197,8 +200,14 @@ const cutTranscriptionEventHandler = async (event: TranscriptionEvent) => {
...
 }
 
+let responseMutex = false;
 const responseReflexEventHandler = async (): Promise<void> => {
   await globalWhisperPromise;
+  if (!responseMutex) {
+    responseMutex = true;
+    return;
+  }
+  responseMutex = false;
```
Comment (yacineMTB) [inline index.ts]: likely regret
instead, responseReflexEventHandler is actually something that should downstream from the transcription event
this way, we can do away with the responsemutex state
also, the responsemutex should actually be an event in the event log, that the transcription event looks for
if it sees one that hasn't been begun, then it kicks off response
does that make sense?

Comment (yacineMTB) [reply index.ts]: this avoids adding state and still solves for "transcription still needs to end for new information that would be useful for response"
you feel me dawg?

Comment (choombaa) [reply index.ts]: I know what you mean, you're right, I'll revise

**Comment by choombaa** [comment]
I split the ResponseReflex into two events
Transcription -> ResponseInput -> Transcription -> ResponseReflex

This works as long as transcription n=2

**Comment by choombaa** [comment]
I thought about making a generic InputEvent but the event searching kind of smells and I don't want to introduce a bad abstraction

**Review by yacineMTB — APPROVED** [review]

**Pull request merged**